sympy/core/add.py

```diff
@@ -613,6 +613,8 @@ def as_numer_denom(self):
         """
         # clear rational denominator
         content, expr = self.primitive()
+        if not isinstance(expr, self.func):
+            return Mul(content, expr, evaluate=False).as_numer_denom()
```

Comment:
Why not just `isinstance(expr, Add)`?

Reply:
I was being literal: if the function type changed, don't handle it here.